src/ray/raylet/actor_registration.cc

```diff
@@ -77,6 +77,53 @@ void ActorRegistration::AddHandle(const ActorHandleID &handle_id,
 
 int ActorRegistration::NumHandles() const { return frontier_.size(); }
 
+void ActorRegistration::RestoreFrontier(const ActorCheckpointDataT &checkpoint_data) {
+  // Restore `execution_dependency_`.
+  execution_dependency_ = ObjectID::from_binary(checkpoint_data.execution_dependency);
+  // Restore `frontier_`.
```

Comment:
We should probably clear `frontier_` and `dummy_objects_` before restoring. Also, we should make sure to release any of the objects in `dummy_objects_` before the restore.